subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -259,15 +259,30 @@ int wifi_utils_parse_scan_bands(char *scan_bands_str, uint8_t *band_map)
 int wifi_utils_parse_scan_ssids(char *scan_ssids_str,
 				char ssids[][WIFI_SSID_MAX_LEN + 1])
 {
+	char parse_str[(WIFI_MGMT_SCAN_SSID_FILT_MAX * (WIFI_SSID_MAX_LEN + 1)) + 1];
```

Comment:
This looks a bit dangerous as the stack usage can grow very large. For the default case where filt_max is 1, there should be no problems, but if the user specifies large value for `WIFI_MGMT_SCAN_SSID_FILT_MAX` then we are in trouble.
Could we limit this to some sane max value, typically SSIDs are not full 32 bytes long so we probably do not need to allocate max length buf here, WDYT?

Reply:
But that's true everywhere we use `WIFI_MGMT_SCAN_SSID_FILT_MAX`, e.g., `cmd_wifi_scan`, IMHO we should just add `WIFI_MGMT_SCAN_SSID_FILT_MAX` range `range 1 4`, more than 4 SSIDs is almost never needed.

Reply:
Ok, we could start with that.